docs/applications/visualisation.py

```diff
@@ -161,4 +163,59 @@ def interactive_view(system, scale=0.5, name=""):
     view.center()
     view._remote_call("setSize", target="Widget", args=["300px", "300px"])
     view._js(tooltip_js)
-    return view
\ No newline at end of file
+    return view
+
+
+def show_stacking_fault(images, ax=None, CNA_color=True, **kwargs):
```

Comment:
These routines should (eventually) be made accessible to end users, not just in the docs

Reply:
I agree, I think I was asking it in previous PR regarding ngl visulasiations as well. At the beginning I thought it will be just a quick fix just for the docs, but it looks to be rather useful for everyday use as well. We have two options:
- make it part of a separate module like `matscipy.visualisation.py` 
- make it part of the `dislocation` and `gamma_surface` modules. 

The latter one has an advantage of passing all the necessary parameters under the hood.

Reply:
I would say that option 2 is preferable. I'd worry that a `visualisation.py` module would be extremely difficult to document, as you would really need to have read all of the rest of the documentation to know about dislocations etc, especially since `show_dislocation` is pretty manual to set up anyway. Having visualisation attached to the class means that you'd only really need formatting arguments, and that it could be used by people who are just playing with the tools available for the first time.

I can also see it being extremely useful for people developing from things like jupyter notebooks, as the visualisation would be extremely accessible, requiring no new imports.

The only question I have is the best way to structure this - I'm happy to do it all as part of this PR, but I'm also conscious of it already being large, and I think it's also a bit messy for me to be making big changes to dislocations from a PR marked as gamma surface. Maybe we port `show_stacking_fault` to the classes in this PR, and have a new PR to port the dislocations side of things?

Reply:
Let's do it in two PRs, keeping this one focussed on gamma surfaces and stacking faults

Reply:
ok, we should go for the second option then and I also agree that it should be another PR since it is much more broader than just gamma surface and stacking faults.

So the idea will be to have visualisation methods embedded as a object methods such as `W_screw.show_dislocation(atoms)` or `W_sf.show_stacking_fault(atoms)`. Do I understand it correctly? It is very similar to what I wanted to do for functions like `plot_vitek()` since for every dislocation system it has different parameters and in this way we can pass them automatically.  

We will probably have to keep the visulalisation imports inside this functions in order not to include it to the default dependencies, but this is just a technical detail.

Reply:
I think a more concise `W_screw.show()` is nicer, but a small detail. And yes, visualisation imports inside of the function sounds good to me, so we don't add loads of extra mandatory requirements (does nglview come with ase!?). I'll work on packaging the stacking fault stuff in this PR now.

Reply:
I have a working version of some nice plotting for the stacking faults, but I needed to move the ovito DXA from `visualisation.py` into the new `utils.py` - @pgrigorev it may be easier to modify the dislocation plotting once this is merged?